drivers/digit7seg/doc.txt

```diff
@@ -0,0 +1,180 @@
+parameters:
+
+Parameter                      | Symbol in ```digit7seg_params.h``` | Default
+:------------------------------|:-----------------------------------|:------------------
+Data pin for a segment         | DIGIT7SEG_PARAM_A_PIN              | GPIO_PIN(1, 11)
+Data pin for b segment         | DIGIT7SEG_PARAM_B_PIN              | GPIO_PIN(1, 12)
+Data pin for c segment         | DIGIT7SEG_PARAM_C_PIN              | GPIO_PIN(1, 15)
+Data pin for d segment         | DIGIT7SEG_PARAM_D_PIN              | GPIO_PIN(1, 13)
+Data pin for e segment         | DIGIT7SEG_PARAM_E_PIN              | GPIO_PIN(1, 14)
+Data pin for f segment         | DIGIT7SEG_PARAM_F_PIN              | GPIO_PIN(0, 23)
+Data pin for g segment         | DIGIT7SEG_PARAM_G_PIN              | GPIO_PIN(0, 21)
+Data pin for dp segment        | DIGIT7SEG_PARAM_DP_PIN             | GPIO_PIN(0, 27)
+Pin to on digit 1              | DIGIT7SEG_PARAM_DIG1_PIN           | GPIO_PIN(1, 2)
+Pin to on digit 2              | DIGIT7SEG_PARAM_DIG2_PIN           | GPIO_PIN(1, 1)
+Pin to on digit 3              | DIGIT7SEG_PARAM_DIG3_PIN           | GPIO_PIN(1, 8)
+Pin to on digit 4              | DIGIT7SEG_PARAM_DIG4_PIN           | GPIO_PIN(0, 13)
+Timer for periodic interrupt   | DIGIT7SEG_PARAM_TIMER              | TIMER_DEV(2)
+Number of digits on the periph | DIGIT7SEG_PARAM_DIGITS             | 4
+
+The default configuration can be overwritten by the application.
+
+Example:
+```
+#define DIGIT7SEG_PARAM_TIMER TIMER_DEV(1)
+#define DIGIT7SEG_PARAM_DIGITS (2)
+...
+#include "digit7seg.h"
+#include "digit7seg_param.h"
+```
+
+Another way to override params is to override DIGIT7SEG_PARAMS that will
+be stocked into digit7seg_params.
+
+@note At least all the data_* pin must be defined, depending of the number of digits set, dig4, dig3, dig2 can be set at GPIO_UNDEF.
+
+Example:
+```
+#define DIGIT7SEG_PARAMS              { .data_a = GPIO(1, 14),   \
+                                        .data_b = GPIO(1, 17),   \
+                                        .data_c = GPIO(1, 15),   \
+                                        .data_d = GPIO(1, 13),   \
+                                        .data_e = GPIO(1, 12),   \
+                                        .data_f = GPIO(0, 23),   \
+                                        .data_g = GPIO(0, 21),   \
+                                        .data_dp = GPIO(1, 11), \
+                                        .dig1 = GPIO(0, 13),  \
+                                        .dig2 = GPIO(1, 8),  \
+                                        .dig3 = GPIO_UNDEF,  \
+                                        .dig4 = GPIO_UNDEF,  \
+                                        .timer = TIMER_DEV(2),    \
+                                        .digits = 2 }
+...
+#include "digit7seg.h"
+#include "digit7seg_param.h"
+```
+
+The ::digit7seg_init function initializes a #digit7seg_t and checks if every parameter is correctly set.
```

Comment:
or maybe you know more than me and this is actually some special doxygen-markdown syntax?

Reply:
It's a special dioxygen ref